runtime/interpreter/interpreter.go

```diff
@@ -1151,6 +1187,14 @@ func (interpreter *Interpreter) declareNonEnumCompositeValue(
 				value.Destructor = destructorFunction
 
 				var self MemberAccessibleValue = value
+				if declaration.CompositeKind == common.CompositeKindAttachment {
+					self = NewEphemeralReferenceValue(interpreter, false, value, interpreter.MustSemaTypeOfValue(value))
+					// set the base to the implicitly provided value, and remove this implicit argument from the list
+					implicitArgumentPos := len(invocation.Arguments) - 1
+					invocation.Base = invocation.Arguments[implicitArgumentPos].(*EphemeralReferenceValue)
+					invocation.Arguments = invocation.Arguments[:implicitArgumentPos]
+					invocation.ArgumentTypes = invocation.ArgumentTypes[:implicitArgumentPos]
```

Comment:
Also, clear out the underlying pointer to the argument value and argument type:

```suggestion
					invocation.Arguments[:implicitArgumentPos] = nil
					invocation.Arguments = invocation.Arguments[:implicitArgumentPos]
					
					invocation.ArgumentTypes[:implicitArgumentPos] = nil
					invocation.ArgumentTypes = invocation.ArgumentTypes[:implicitArgumentPos]
```